feat: add webhookURL helm option

golpa opened the pull request:
**What this PR does / why we need it**:
Adds a `webhookURL` option to the helm chart to allow for an alternate way for the Kubernetes API server to access the webhooks.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #2719 

<!--
**Is the PR title following semantic convention?**
Please refer to [Semantic types](https://github.com/open-policy-agent/gatekeeper/blob/master/.github/semantic.yml) to view accepted title convention to satisfy this status check.  
-->

<!--
**Are you making changes to Gatekeeper Helm chart?**
Please refer to [Contributing to Helm Chart](https://open-policy-agent.github.io/gatekeeper/website/docs/help#contributing-to-helm-chart) for modifying the Helm chart.
-->

<!--
**Are you making changes to Gatekeeper docs?**
Please see [Contributing to Docs](https://open-policy-agent.github.io/gatekeeper/website/docs/help#contributing-to-docs) 
-->

**Special notes for your reviewer**:

sozercan commented:
@golpa thanks for the PR! can you add this to helm parameters list too? https://github.com/open-policy-agent/gatekeeper/blob/master/cmd/build/helmify/static/README.md?plain=1#LL63C1-L63C1

sozercan commented on cmd/build/helmify/replacements.go:
```diff
@@ -123,6 +123,15 @@ var replacements = map[string]string{
   {{- end }}`,
 
+	"HELMSUBST_MUTATING_WEBHOOK_CLIENT_CONFIG: \"\"": `{{- if .Values.webhookURL }}
+    url: https://{{ .Values.webhookURL }}/v1/mutate
```
perhaps different values for `validatingWebhookURL` and `mutatingWebhookURL` in case someone needs them to be separate?

golpa replied:
I've made the change to support those two URLs separately.

golpa commented:
I've updated the `values.yaml` file with the default of `null` for `validatingWebhookURL` and `mutatingWebhookURL`. I've also added a description to the `README.md` for each of those options.

sozercan commented on manifest_staging/charts/gatekeeper/templates/gatekeeper-mutating-webhook-configuration-mutatingwebhookconfiguration.yaml:
```diff
@@ -15,10 +15,14 @@ webhooks:
   - v1
   - v1beta1
   clientConfig:
+    {{- if .Values.webhookURL }}
```
these need to be regenerated. can you run `make manifests` please?

golpa replied:
oops forgot about that. Ran and pushed changes

sozercan reviewed (approved):
LGTM

maxsmythe reviewed (approved):
LGTM

ritazh reviewed (approved):
LGTM
